### PR TITLE
feat(gatsby-plugin-gatsby-cloud): request customer feedback

### PR DESCRIPTION
**Pull request by leithonenglish**
Adds functionality to collect feedback from customers. Randomly ask users to share feedback after interacting with the Indicator and a certain period of time has passed

- Displays the prompt once the 30 days have passed and the user has interacted with the Indicator at least `10` times
- If dismissed, don't display for another 30 days
- If clicked, don't display for another 30 days
- Share these settings across Preview so users aren't bombarded with feedback requests. Added cookies to track the feedback loop
- Updated tooltip to ensure that tooltip is not displayed and is not just invisible
- Added tip to the tooltip

**Feedback tooltip prompt**
Screenshot 2022-01-20 at 12 53 37 PM

**Tooltip with new tip**
Screenshot 2022-01-20 at 12 51 38 PM

### CODE REVIEW

**Comment by TylerBarnes** [comment]
PR titles in this repo should be made using the [Conventional Commits spec](https://www.conventionalcommits.org/en/v1.0.0/). Can you update your title to something like `feat(gatsby-plugin-gatsby-cloud): Add customer feedback modal`? Thanks 😄

File: packages/gatsby-plugin-gatsby-cloud/package.json
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/package.json:55]: I think we can leave this out of this PR since TS conversion will come in another PR 👍

File: packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js:73]: This is much nicer

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/IndicatorButton.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/IndicatorButton.js:25]:
```suggestion
      if (typeof onMouseEnter === `function`) {
```

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/IndicatorButton.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/IndicatorButton.js:33]:
```suggestion
    if (active && hoverable && typeof onClick === `function`) {
```

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js:103]: Might be good to put this at the top of the file as `const FEEDBACK_URL =`

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js:207]: No need to change this but fyi you can make it more concise like `buildStatusActions?.[buildStatus] || null`

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js:208]:
```suggestion
    if (typeof buildStatusAction === `function`) {
```

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js:204]: I find code a lot easier to scan through if you put a blank line between blocks of code and after variables. Not a big deal but worth mentioning :)

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js:82]:
```suggestion

```

**Review by TylerBarnes — CHANGES_REQUESTED** [review]
Nice work :) I left some comments about code style generally, nothing major. Pending these changes and a review/changes from @pvorozhe and @veryspry this looks good to go! 💪

File: packages/gatsby-plugin-gatsby-cloud/utils/index.js
Comment (TylerBarnes) [inline packages/gatsby-plugin-gatsby-cloud/utils/index.js:3]: @leithonenglish this file should be removed

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/GatsbyIndicatorButton.js
Comment (pvorozhe) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/GatsbyIndicatorButton.js:18]: This.. will actually break thing si think. we need to change it to always returning 

```
return {
        active: true,
      }
```

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/GatsbyIndicatorButton.js
Comment (pvorozhe) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/GatsbyIndicatorButton.js:14]: this seems redundant, the gatsby icon should just always return active: true

File: packages/gatsby-plugin-gatsby-cloud/src/components/buttons/GatsbyIndicatorButton.js
Comment (pvorozhe) [inline packages/gatsby-plugin-gatsby-cloud/src/components/buttons/GatsbyIndicatorButton.js:20]: will always be null

File: .eslintignore
Comment (LekoArts) [inline .eslintignore:30]: These changes seem unrelated to the PR?

File: packages/gatsby-plugin-gatsby-cloud/.gitignore
Comment (LekoArts) [inline packages/gatsby-plugin-gatsby-cloud/.gitignore:2]: Also seems unrelated

File: packages/gatsby-plugin-gatsby-cloud/src/models/components/index.ts
Comment (LekoArts) [inline packages/gatsby-plugin-gatsby-cloud/src/models/components/index.ts:1]: If you're adding TypeScript files you'll need to adjust the toolchain / build command. I think you can remove the TypeScript files?

**Review by LekoArts — CHANGES_REQUESTED** [review]

File: packages/gatsby-plugin-gatsby-cloud/src/models/components/index.ts
Comment (leithonenglish) [reply packages/gatsby-plugin-gatsby-cloud/src/models/components/index.ts:1]: I need them for the enum

File: packages/gatsby-plugin-gatsby-cloud/.gitignore
Comment (leithonenglish) [reply packages/gatsby-plugin-gatsby-cloud/.gitignore:2]: Yeah we added that to make sure it captures the right js files

File: .eslintignore
Comment (leithonenglish) [reply .eslintignore:30]: I'm not sure about how that got in there. I didn't make that change

File: packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js
Comment (DanielSLew) [inline packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js:109]: I kind of like the `cleanup` name there, it describes what it's doing.

File: .eslintignore
Comment (LekoArts) [reply .eslintignore:30]: You'll need to revert those then

File: packages/gatsby-plugin-gatsby-cloud/src/models/components/index.ts
Comment (LekoArts) [reply packages/gatsby-plugin-gatsby-cloud/src/models/components/index.ts:1]: But I don't see them used anywhere? They'll also not be built as babel/tsc is not converting them

File: packages/gatsby-plugin-gatsby-cloud/.gitignore
Comment (LekoArts) [reply packages/gatsby-plugin-gatsby-cloud/.gitignore:2]: If we remove the TypeScript `models` files the file can stay the same

File: .eslintignore
Comment (leithonenglish) [reply .eslintignore:30]: Cool

Comment (leithonenglish) [reply .eslintignore]: Reversed the change

**Review by TylerBarnes — APPROVED** [review]

**Comment by TylerBarnes** [comment]
Looks good to go once the folks who asked for changes approve 😄 🚀 nice work @leithonenglish !

File: packages/gatsby-plugin-gatsby-cloud/index.js
Comment (LekoArts) [inline packages/gatsby-plugin-gatsby-cloud/index.js:2]: Please revert changing that file

File: packages/gatsby-plugin-gatsby-cloud/package.json
Comment (LekoArts) [inline packages/gatsby-plugin-gatsby-cloud/package.json:55]: We don't need source-maps here

File: packages/gatsby-plugin-gatsby-cloud/package.json
Comment (LekoArts) [inline packages/gatsby-plugin-gatsby-cloud/package.json:59]: Also no source-maps

File: packages/gatsby-plugin-gatsby-cloud/src/__tests__/gatsby-browser.js
Comment (LekoArts) [inline packages/gatsby-plugin-gatsby-cloud/src/__tests__/gatsby-browser.js:215]: Please clarify here and/or in the PR description:

- What "not working as expected" means
- What changes would be necessary to fix this
- How bad skipping these tests is for coverage
- If they can't be fixed in this test, when the follow-up PR would be put up

File: packages/gatsby-plugin-gatsby-cloud/src/components/index.d.ts
Comment (LekoArts) [inline packages/gatsby-plugin-gatsby-cloud/src/components/index.d.ts:1]: Why was the `export` removed?

File: packages/gatsby-plugin-gatsby-cloud/src/index.js
Comment (LekoArts) [inline packages/gatsby-plugin-gatsby-cloud/src/index.js:1]: This file can be deleted as we have it in the root already

**Review by LekoArts — CHANGES_REQUESTED** [review]
Please format your PR description to our template, so in broad strokes explain your changes, and lay out some changes you made and why, e.g. for skipping some tests. Linking to our internal tools won't be enough as in some years we could have switched and we need a paper trace of changes in GitHub, too.

You'll also want to fix the merge conflict, format the PR title to conventional commits and add the appropriate label to this PR.

Thanks! We're nearly done 🎉

File: packages/gatsby-plugin-gatsby-cloud/src/index.js
Comment (leithonenglish) [reply packages/gatsby-plugin-gatsby-cloud/src/index.js]: Done!

File: packages/gatsby-plugin-gatsby-cloud/src/components/index.d.ts
Comment (leithonenglish) [reply packages/gatsby-plugin-gatsby-cloud/src/components/index.d.ts]: Done!

File: packages/gatsby-plugin-gatsby-cloud/src/__tests__/gatsby-browser.js
Comment (leithonenglish) [reply packages/gatsby-plugin-gatsby-cloud/src/__tests__/gatsby-browser.js]: Add more detail to the not let me know if more is needed.

**Review by TylerBarnes — APPROVED** [review]
It looks good to me 👍 also looks like all changes were addressed so I'll approve/merge. @LekoArts let us know if I missed one of your change requests and we can open an additional PR if needed

**Pull request merged**
